### PR TITLE
Fixed bug in forced Search: _addCacheEntry paramater mismatch

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

This PR started as a bugfix. But while working on it, I took the opportunity to also make it unicode compliant, as where working on that anyway. But then I noticed we just merged the stand-providers, which had also already changes to tvcache. That's when I decided to cleanup the file.

Fixed bug in forced Search: _addCacheEntry paramater mismatch called from GenericProvider.py.
Error shows with ForcedSearch. Maybe also backlogs.

* Cleaned up tvcache.py (format, single quotes and pep8 snake_case nameings where safe.)
* Enabled import unicode_literals for tvcache.py

### CODE REVIEW

File: sickrage/providers/GenericProvider.py [hunk 1]
```diff
@@ -264,7 +266,7 @@ def find_search_results(self, show, episodes, search_mode, forced_search=False,
                 logger.log(u'Adding item from search to cache: %s' % title, logger.DEBUG)
                 # pylint: disable=protected-access
                 # Access to a protected member of a client class
-                ci = self.cache._addCacheEntry(title, url, seeders, leechers, size, pubdate, hash, parse_result=parse_result)
+                ci = self.cache._addCacheEntry(title, url, seeders, leechers, size, pubdate, torrent_hash)
```
Comment (p0psicles) [inline sickrage/providers/GenericProvider.py]: This is the actual bug. addCacheEntry doesn't provide paramater parse_result anymore.

**Comment by p0psicles** [comment]
@medariox can you mark, where you think we don't need the b'' ?

File: sickbeard/tvcache.py [hunk 2]
```diff
@@ -38,62 +40,66 @@ def __init__(self, providerName):
...
-            logger.log(u"Creating UNIQUE URL index for {}".format(providerName), logger.DEBUG)
-            self.action("CREATE UNIQUE INDEX IF NOT EXISTS idx_url_" + providerName + " ON [" + providerName + "] (url)")
+            logger.log(b'Creating UNIQUE URL index for {0}'.format(providerName), logger.DEBUG)
+            self.action(b'CREATE UNIQUE INDEX IF NOT EXISTS idx_url_{0}  ON [{1}] (url)'.
+                        format(providerName, providerName))
 
             # add release_group column to table if missing
             if not self.hasColumn(providerName, 'release_group'):
-                self.addColumn(providerName, 'release_group', "TEXT", "")
+                self.addColumn(providerName, 'release_group', 'TEXT', '')
```
Comment (medariox) [inline sickbeard/tvcache.py]: Here, I was wonder if we need it. But I'm quite sure we don't.

File: sickbeard/tvcache.py [hunk 3]
```diff
@@ -38,62 +40,66 @@ def __init__(self, providerName):
...
+                    self.action(b'DELETE FROM [{provider_name}] WHERE url = ?'.format(provider_name=providerName), [cur_dupe[b'url']])
 
             # remove wrong old index
-            self.action("DROP INDEX IF EXISTS idx_url")
+            self.action(b'DROP INDEX IF EXISTS idx_url')
 
             # add unique index to prevent further dupes from happening if one does not exist
-            logger.log(u"Creating UNIQUE URL index for {}".format(providerName), logger.DEBUG)
-            self.action("CREATE UNIQUE INDEX IF NOT EXISTS idx_url_" + providerName + " ON [" + providerName + "] (url)")
+            logger.log(b'Creating UNIQUE URL index for {0}'.format(providerName), logger.DEBUG)
```
Comment (p0psicles) [inline sickbeard/tvcache.py]: You mean this one? Else I don't know where I need to remove the b''. The log is a mistake indeed.

Comment (medariox) [inline sickbeard/tvcache.py]: No, look at my comment above.

**Comment by medariox** [comment]
Approved

**Comment by labrys** [comment]
Approved!

**Pull request merged**